### PR TITLE
OCPCLOUD-2014: Switch Azure credentials request to use explicit permissions

### PR DESCRIPTION
**Pull request by dlom**
xref: [OCPCLOUD-2014](https://issues.redhat.com//browse/OCPCLOUD-2014)

related CCO PR: https://github.com/openshift/cloud-credential-operator/pull/556

### CODE REVIEW

File: install/0000_30_machine-api-operator_00_credentials-request.yaml [hunk 1]
```diff
@@ -82,8 +82,46 @@ spec:
   providerSpec:
     apiVersion: cloudcredential.openshift.io/v1
     kind: AzureProviderSpec
-    roleBindings:
-    - role: Contributor
+    permissions:
+    - Microsoft.Compute/availabilitySets/delete
+    - Microsoft.Compute/availabilitySets/read
+    - Microsoft.Compute/availabilitySets/write
+    - Microsoft.Compute/disks/delete
```
Comment (dlom) [inline install/0000_30_machine-api-operator_00_credentials-request.yaml:89]: Based on the comments [here](https://github.com/openshift/machine-api-provider-azure/blob/97ea1221e4ca999a161ddee4d1e075880fcbad67/pkg/cloud/azure/services/disks/disks.go#L34), I have only added the permission for `disks/delete`.  I'm not sure yet whether the `read` and `write` permissions are required.

**Comment by JoelSpeed** [comment]
Azure tests passed, what is blocking this form merging?

**Comment by abutcher** [comment]
@dlom Could we add `e2e-azure-manual-oidc` to machine-api-operator in openshift/release to test with that job

**Comment by JoelSpeed** [comment]
```
E0720 18:19:19.901194       1 actuator.go:84] Machine error: failed to reconcile machine "ci-op-8zsvfysh-42948-ncf5b-worker-centralus1-ltslw": network.InterfacesClient#CreateOrUpdate: Failure sending request: StatusCode=403 -- Original Error: Code="LinkedAuthorizationFailed" Message="The client 'eba1a2a6-b0c1-4f9a-992c-ee1cbe745fd6' with object id 'eba1a2a6-b0c1-4f9a-992c-ee1cbe745fd6' has permission to perform action 'Microsoft.Network/networkInterfaces/write' on scope '/subscriptions/d38f1e38-4bed-438e-b227-833f997adf6a/resourceGroups/ci-op-8zsvfysh-42948/providers/Microsoft.Network/networkInterfaces/ci-op-8zsvfysh-42948-ncf5b-worker-centralus1-ltslw-nic'; however, it does not have permission to perform action 'Microsoft.Network/virtualNetworks/subnets/join/action' on the linked scope(s) '/subscriptions/d38f1e38-4bed-438e-b227-833f997adf6a/resourceGroups/ci-op-8zsvfysh-42948/providers/Microsoft.Network/virtualNetworks/ci-op-8zsvfysh-42948-ncf5b-vnet/subnets/ci-op-8zsvfysh-42948-ncf5b-worker-subnet' or the linked scope(s) are invalid."
```
From the machine controller, appears another permission required?

**Comment by dlom** [comment]
@JoelSpeed this one is finally ready I think

**Comment by JoelSpeed** [comment]
Azure test failures look unrelated 
/test e2e-azure-operator
/test e2e-azure-ovn

**Comment by dlom** [comment]
/assign @JoelSpeed 
All azure-related e2e is green

**Review by elmiko — COMMENTED** [review]
i'm no azure permissions expert, but this generally makes sense to me

/lgtm

**Comment by dlom** [comment]
@JoelSpeed based on comments made on the card, it doesn't look like the issue encountered by QE is related to this change

**Comment by JoelSpeed** [comment]
Another comment was left this morning, can you please review the new comment from QE

**Comment by jstuever** [comment]
/test e2e-azure-manual-oidc
Replication delay again.

**Comment by JoelSpeed** [comment]
/hold cancel

Now QE approved

**Pull request merged**